clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerUtils.java

```diff
@@ -211,4 +226,17 @@ else if (t instanceof KafkaException)
             throw new TimeoutException(e);
         }
     }
+
+    public static boolean maybeOverrideEnableAutoCommit(ConsumerConfig config) {
+        Optional<String> groupId = Optional.ofNullable(config.getString(CommonClientConfigs.GROUP_ID_CONFIG));
+        boolean enableAutoCommit = config.getBoolean(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG);
+        if (!groupId.isPresent()) { // overwrite in case of default group id where the config is not explicitly provided
```

Comment:
Did we add a check to invalidate empty groupId?  https://issues.apache.org/jira/browse/KAFKA-14438

Reply:
I was also wondering if this should should happen in the ConsumerConfig

Reply:
I didn't add it in this PR, no. The Jira refers to doing that in the "next major release," so that would imply we should hold off and handle it outside this PR.